prow/cmd/deck/static/common/abort.ts

```diff
@@ -1,41 +1,76 @@
 import {ProwJobState} from "../api/prow";
 import {showAlert, showToast, State} from "./common";
 
-export function createAbortProwJobIcon(state: ProwJobState, prowjob: string, csrfToken: string): HTMLElement {
+export function createAbortProwJobIcon(modal: HTMLElement, parentEl: Element, job: string, state: ProwJobState, prowjob: string, csrfToken: string): HTMLElement {
   const url = `${location.protocol}//${location.host}/abort?prowjob=${prowjob}`;
-  const button = document.createElement('button');
-  button.classList.add('mdl-button', 'mdl-js-button', 'mdl-button--icon');
-  button.innerHTML = '<i class="icon-button material-icons" title="Cancel this job" style="color: gray">cancel</i>';
+  const abortButton = document.createElement('button');
+  abortButton.classList.add('mdl-button', 'mdl-js-button', 'mdl-button--icon');
+  abortButton.innerHTML = '<i class="icon-button material-icons" title="Cancel this job" style="color: gray">cancel</i>';
 
+  const closeModal = (): void => {
+    modal.style.display = "none";
+    // Resets modal content. If removed, elements will be concatenated, causing duplicates.
+    parentEl.classList.remove('abort-content', 'rerun-content');
+    parentEl.innerHTML = '';
+  };
+  window.onkeydown = (event: any) => {
+    if (event.key === "Escape") {
+      closeModal();
+    }
+  };
+  window.onclick = (event: any) => {
+    if (event.target === modal) {
+      closeModal();
+    }
+  };
   if (state !== State.TRIGGERED && state !== State.PENDING) {
-    button.innerHTML = `<i class="icon-button material-icons" title="Can't abort job in ${state} state" style="color: lightgray">cancel</i>`;
-    button.classList.remove('icon-button');
-    button.disabled = true;
+    abortButton.innerHTML = `<i class="icon-button material-icons" title="Can't abort job in ${state} state" style="color: lightgray">cancel</i>`;
+    abortButton.classList.remove('icon-button');
```

Comment:
Does it ever have that class? It only looks like the nested `<i>` inside does.

Reply:
Good catch there, you are right! This has been fixed!